src/patchkernel/patch_kernel.cpp

```diff
@@ -5635,6 +5635,147 @@ void PatchKernel::getInterfaceVertexCoordinates(long id, std::array<double, 3> *
 	getElementVertexCoordinates(interface, coordinates);
 }
 
+/*!
+   Check if an interface  is intersected by a given plane. Calculate the intersection
+   as a segment defined by two points and the part of the interface belonging to the subspace
+   opposite to the one pointed by the given normal. If the interface is not intersected,
+   an empty polygon is returned. 
+
+   Interfaces laid on the plane are not considered as intersected ones.
```

Comment:
I still don't like this, because it seems counter intuitive.

Reply:
If the interface lays on the plane, the intersection becomes the face itself. However, this function is designed to provide a segment as an intersection. Also, if we consider this case as an intersection, we have to consider others as well, like when the interface has only one of its vertices on the plane. In this case the intersection is just this node. Finally, in a case when an edge is on the surface, the user won't be able to understand without further computations if the intersection is a "real" one or just the edge of the interface, because in both cases the intersection will be represented by a segment. Are you sure that you prefer this behavior?